index.js

```diff
@@ -149,42 +190,60 @@ const line = []
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 1", "color: red")
+console.log("%cScope & Closures - Question 1", "color: red")
+
+var count = 0
+function takeATicketNumber(array){
+  count += 1;
+  array.push(count);
+  return `Welcome. You are ticket number ${count}`;
+}
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 1`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 1`
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 2`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 2`
 
-// console.log(nowServing(line))
-// // => `Currently serving 1.`
+console.log(nowServing(line))
+// => `Currently serving 1.`
 
-// console.log(nowServing(line))
-// // => `Currently serving 2.`
-// console.log("%c----------", "color: red") 
+console.log(nowServing(line))
+// => `Currently serving 2.`
+console.log("%c----------", "color: red") 
 
-// console.log(takeATicketNumber(line))
-// // => `Welcome. You are ticket number 3`
+console.log(takeATicketNumber(line))
+// => `Welcome. You are ticket number 3`
 
 
 // ***** Scope & Closures - Question 2 *****
 
 
 // *** Uncomment the lines below to test
-// console.log("%cScope & Closures - Question 2", "color: red")
+console.log("%cScope & Closures - Question 2", "color: red")
 
-// const newLine = []
-// const takeATicketNumberFunc = ticketNumberGeneratorFunc()
-// console.log(takeATicketNumberFunc(newLine))
-// // => `Welcome. You are ticket number 1`
+//Couldn't quite figure out how to get this one to work
+
+function ticketNumberGeneratorFunc(){
+  count = 0
+  function takeATicketNumber(array){
+    count ++
+    array.push(count)
+    return `Welcome. You are ticket number ${count}`
+  }
+  return takeATicketNumber
+}
```

Comment:
This is close! On line 228, you should create a new variable with `let` instead of using the `count` variable from the global scope:

```js
function ticketNumberGeneratorFunc(){
  let count = 0
  function takeATicketNumber(array){
    count ++
    array.push(count)
    return `Welcome. You are ticket number ${count}`
  }
  return takeATicketNumber
}
```